server/server/data_loader/arangoload.py

```diff
@@ -183,6 +194,100 @@ def process_division_files(
     docs.append({'uid': 'orphan', '_key': 'orphan'})
 
 
+def process_names_files(
+        names_files: List[Path],
+        root_languages: Dict[str, str],
+        super_extra_info:  Dict[str, Dict[str, str]],
+        text_extra_info:  Dict[str, Dict[str, str]]
+) -> List[dict]:
+    """
+    Method for processing name files from sc-data/structure/name
+
+    Args:
+        names_files - list of name Path objects to files from name folder
+        root_languages - parsed data from language.json
+        super_extra_info - parsed data from super_extra_info.json
+        text_extra_info - parsed data from text_extra_info.json
+
+    Returns:
+        list of processed data
+    """
+    docs = []
+    names_files.sort(key=lambda path: len(path.parts))
+    for name_file in names_files:
+        entries: Dict[str, str] = json_load(name_file)
+        for uid, name in entries.items():
+            if not name or type(name) != str:
+                continue
+            extra_info = super_extra_info if uid in super_extra_info else text_extra_info
+            entry = {
+                'uid': uid,
+                '_key': uid,
+                'name': name,
+                'root_lang': root_languages.get(uid, None),
+                'volpage': extra_info.get(uid, {}).get('volpage', None),
+                'biblio_uid': extra_info.get(uid, {}).get('biblio_uid', None),
+                'acronym': extra_info.get(uid, {}).get('acronym', None)
+            }
+
+            docs.append(entry)
+    return docs
+
+
+def parse_tree_recursive(element: Dict[str, list]) -> List[Dict[str, str]]:
+    """
+    Method to parse a subtree from tree files and super-tree.json file
+
+    Args:
+        element - dict with one key as uid and list of elements with the
+            same structure as value
+    """
+    edges = []
+    for name, content in element.items():
+        for item in content:
+            if type(item) == dict:
+                edges.extend(
+                    [{'_from': name, '_to': subgroup_name} for subgroup_name in item.keys()]
+                )
+                edges.extend(parse_tree_recursive(item))
+            else:
+                edges.append({
+                    '_from': name,
+                    '_to': item,
+                })
```

Comment:
The same here you can use map function.

Reply:
Well actually we can, yes, but personally I don't think it's the "Pythonic way" to use the `map` function here. I'd rather use `map` if I had to change every element of the incoming list in some way and want a list/generator of changed objects in the output, although even then I'd consider using list comprehensions or generator expressions (though again - it all depends on the case).
but personally I think that in this case regular loops are more readable and don't cause any discomfort at all